File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-8.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    goodNodes(root) {
        let res = 0;

        function dfs(node, max){
            if(!node) return;

            let mmax = max;

            if(node.val >= max){
                mmax = node.val;
                res++;
            }

            if(node.left) dfs(node.left, mmax);
            if(node.right) dfs(node.right, mmax);
        }

        dfs(root, -Infinity);
        return res;
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-8.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        
        lst = []

        def dfs(node):

            if not node:
                return
            
            left = dfs(node.left)
            lst.append(node.val)
            right = dfs(node.right)

            return 
        
        dfs(root)
        
        return lst[k - 1]